src/app/AttributePathParams.h

```diff
@@ -29,6 +29,11 @@ enum class AttributePathFlags : uint8_t
     kListIndexValid = 0x02,
 };
 
+/**
+ *
+ * @brief This is used to record attribute path, and can pass attribute path info between IM and ember library.
```

Comment:
We don't really have an "ember library"....   Just take that part out?

Also, this seems like an odd data structure in that it can record some attribute paths, but not all.  It can reference an entire attribute, or a list item within a list-valued attribute, but that's it.  Please document what this thing can actually represent.

Reply:
Maybe the right thing to do is to document that this is a struct that represents a subset of the spec concept of "AttributePath" and clearly explain what subset that is?